fs/exec.c

```diff
@@ -1558,6 +1558,8 @@ static void bprm_fill_uid(struct linux_binprm *bprm)
 		 !kgid_has_mapping(bprm->cred->user_ns, gid))
 		return;
 
+	printk(KERN_ERR "linux-hardened: %s executed by the uid/euid:%u/%u just used a SUID\n", current->comm, current_uid(), current_euid());
```

Comment:
I understand that one may want to know the usage of suid binaries for potential correlation of forensic logs after exploitation happened, but this surely is not a `KERN_ERR` but an `KERN_INFO` or `KERN_WARNING` at best.
However, I think a more appropriate way would be the use of the audit subsystem, even when its a bit annoying to extend.

Reply:
Thank you for your prompt return.
I confess I hesitated between KERN_ERR and KERN_WARNING :)

Let's say that with KERN_ERR it's more visible and quicker to read the whole via "journalctl -p err -b". But, you are free to adap this according to your preference ;)

For me, this code has the big advantage to log SUIDs easily by default in order to understand how the system works and also, as you have noticed, to be able to investigate after exploitation.

With this, a sysadmin can record SUID warnings for a standard day and automatically compare them every day. If they do not match, is that there's been some unusual activity.

This code is just keep it stupid simple but it's practical and effective.